tests/subsys/logging/log_syst/src/main.c

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) 2022 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Test log message
+ */
+
+#include <zephyr.h>
+#include <logging/log.h>
+#include "mock_backend.h"
+#include <sys/printk.h>
+#include <logging/log_backend.h>
+#include <logging/log_backend_std.h>
+#include <logging/log_ctrl.h>
+#include <logging/log_output.h>
+#include <tc_util.h>
+#include <stdbool.h>
+#include <ztest.h>
+#include <stdlib.h>
+
+#define TEST_MSG_0 "0 args"
+
+MOCK_LOG_BACKEND_DEFINE(backend1, false);
+
+#define LOG_MODULE_NAME test
+LOG_MODULE_REGISTER(LOG_MODULE_NAME, LOG_LEVEL_DBG);
+
+/**
+ * @brief Testcase to validate that the mock backend would use the expected
+ * processing function from the function pointer table format_table
+ */
+void test_log_syst_format_table_selection(void)
+{
+
+#ifdef CONFIG_LOG_MIPI_SYST_ENABLE
+	uint32_t test_log_type_syst = LOG_OUTPUT_SYST;
+
+	log_format_func_t test_log_output_func = log_format_func_t_get(test_log_type_syst);
+
+	zassert_equal_ptr(test_log_output_func, log_output_msg2_syst_process,
+					"Correct Function pointer for SYST log\n"
+					"log format was not selected %x vs %x",
+					test_log_output_func, log_output_msg2_syst_process);
+
+#elif CONFIG_LOG_BACKEND_MOCK_OUTPUT_DEFAULT == LOG_OUTPUT_TEXT
+
+	uint32_t test_log_type_text = LOG_OUTPUT_TEXT;
+	log_format_func_t test_log_output_func = log_format_func_t_get(test_log_type_text);
+
+	zassert_equal_ptr(test_log_output_func, log_output_msg2_process,
+						"Function pointer for TEXT log format was not selected");
+
+#endif
+
+}
+
+#ifdef CONFIG_LOG_MIPI_SYST_ENABLE
+/* Testcase to validate the SYST output of log data */
+void test_log_syst_data(void)
+{
+	LOG_DBG("Debug message example, %d", 1);
+
+	const char *type = "72";
+	const char *optional_flags = "0A";
+	const char *module_id = "00";
+	const char *sub_type;
+#ifdef CONFIG_64BIT
+	sub_type = "0C";
+#else
+	sub_type = "0B";
+#endif
+	const char *payload =
+	"4465627567206D657373616765206578616D706C652C2025640001000000";
+
+	validate_msg(type, optional_flags, module_id, sub_type, payload);
+}
+
+/* Testcase to validate the SYST output of data with multiple arguments */
+void test_log_syst_data_multiple_args(void)
+{
+	LOG_DBG("Debug message example, %d, %d, %d", 1, 2, 3);
+
+	const char *type = "72";
+	const char *optional_flags = "0A";
+	const char *module_id = "00";
+	const char *sub_type;
+#ifdef CONFIG_64BIT
+	sub_type = "0C";
+#else
+	sub_type = "0B";
+#endif
+	const char *payload_body =
+	"4465627567206D657373616765206578616D706C652C2025642C2025642C202564000";
+	const char *payload_params =
+	"10000000200000003000000";
+	char *payload = malloc(strlen(payload_body)+strlen(payload_params)+1);
```

Comment:
why is this needed?

Reply:
Removed this. Was doing this earlier to address the checkpatch warnings treated as error for long string. Have now used the macro instead.